runtime/doc/provider.txt

```diff
@@ -178,6 +178,29 @@ If `cache_enabled` is |TRUE| then when a selection is copied, Nvim will cache
 the selection until the copy command process dies. When pasting, if the copy
 process has not died, the cached selection is applied.
 
+Instead of using external tools the provider can also use vimscript functions
+>
+    let g:clipboard = {
+          \   'name': 'custom',
+          \   'copy': {
+          \      '+': {lines, regtype -> extend(g:, {'dummy_clipboard_plus': [lines, regtype]}) },
+          \      '*': {lines, regtype -> extend(g:, {'dummy_clipboard_star': [lines, regtype]}) },
+          \    },
+          \   'paste': {
+          \      '+': {-> get(g:, 'dummy_clipboard_plus', [])},
+          \      '*': {-> get(g:, 'dummy_clipboard_star', [])},
+          \   },
+          \ }
+
+The copy function is meant to place a list of lines into the clipboard and
+store a register type specification along with the data.
```

Comment:
It could be worth mentioning the register types: `v` characterwise, `V` linewise and `b` blockwise.

Reply:
I added a  link to `|setreg()|` when merging.

Reply:
ooops I just push a change for this